web-console/src/dialogs/spec-dialog/__snapshots__/spec-dialog.spec.tsx.snap

```diff
@@ -1,6 +1,187 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`spec dialog matches snapshot 1`] = `
+exports[`spec dialog matches snapshot no initSpec 1`] = `
```

Comment:
What's all this new stuff for and what are the `XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX` about?

Reply:
The `.snap` files are snapshots, they are unit tests that virtually render the UI and then take a snapshot of the dom.
The `XXXXXXXXXXXX` is some part of the internals of the AceEditor this is the 3rd party component we use to give the IDE like editor for the JSON. What does it do? No idea, but we will know if it changes!

Reply:
Oh, interesting…

Do we bundle this with the released app or is it skipped? I didn't realize it was a test file at first, because it's in the same `src` path that the main code is in.

Reply:
none of the tests (and test aids like this one) are included in the bundle because none of the code depends on any of the test code (if it did then it would not compile because tests run in a different environment `jest` than the prod code `node/dom`).

In general the web console follows a standard where every class (or functional component) lives in a file next to the style and the test

![image](https://user-images.githubusercontent.com/177816/89696768-39096100-d8ce-11ea-8f88-db213f512c5a.png)

... this has been the case from day one (of the unified console) and is pretty idiomatic of the modern web-dev world.

The tests are separated from the code with file extensions `*.spec.*` vs location

Reply:
TIL